tests/test_completion.py

```diff
@@ -903,3 +903,140 @@ def test_conversion_of_edit_range_to_text_edit_when_it_includes_insert_replace_f
+    ) -> None:
+        item = format_completion(
+            payload, index=0, can_resolve_completion_items=False, session_name='abc', item_defaults={}, view_id=0)
+        self.assertEquals(item.trigger, trigger)
+        self.assertEquals(item.annotation, annotation)
+        self.assertEquals(item.details, details)
+        self.assertEquals(item.flags, flags)
+
+    def test_label(self) -> None:
+        self._verify_completion(
+            {
+                "label": "banner?",
+            },
+            trigger='banner?',
+        )
+
+    def test_detail(self) -> None:
+        self._verify_completion(
+            {
+                "detail": "typescript",
+                "label": "readConfigFile",
+            },
+            trigger='readConfigFile',
+            annotation='typescript'
+        )
+
+    def test_label_details(self) -> None:
+        self._verify_completion(
+            {
+                "label": "banner?",
+                "labelDetails": {
+                    "detail": "()"
+                },
+            },
+            trigger='banner?()',
+        )
+
+    def test_filter_text_1(self) -> None:
+        self._verify_completion(
+            {
+                "filterText": "banner",
+                "label": "banner?",
+            },
+            trigger='banner?',
+        )
+
+    def test_filter_text_2(self) -> None:
+        self._verify_completion(
+            {
+                "filterText": ".$attrs",
+                "label": "$attrs",
+            },
+            trigger='.$attrs',
+            details='$attrs'
+        )
+
+    def test_filter_text_3(self) -> None:
+        self._verify_completion(
+            {
+                "filterText": "import { readConfigFile$1 } from 'typescript';",
+                "label": "readConfigFile",
+            },
+            trigger="import { readConfigFile$1 } from 'typescript';",
+            details='readConfigFile'
+        )
+
+    def test_filter_text_4(self) -> None:
+        # See the `test_filter_text_is_not_a_prefix_of_label` test above and
+        # also https://github.com/sublimelsp/LSP/issues/771
+        # This is probably a silly server behavior that we probably shouldn't need to support?
+        self._verify_completion(
+            {
+                "label": "Implement all members",
+                "filterText": "e",
+            },
+            trigger='e',
+            details='Implement all members'
+        )
+
+    def test_filter_text_and_label_details_1(self) -> None:
+        self._verify_completion(
+            {
+                "filterText": "banner",
+                "label": "banner?",
+                "labelDetails": {
+                    "detail": "()"
+                },
+            },
+            trigger='banner?()',
+        )
+
+    def test_filter_text_and_label_details_2(self) -> None:
+        self._verify_completion(
+            {
+                "filterText": "banner?()",
```

Comment:
Just by looking at this test, 
I do not know if `filterText` will be the trigger or `label+labelDetails`

I would change `filterText` to "ban" just so it is more obvious that `label+labelDetails` will be the trigger...
```suggestion
                "filterText": "ban",
```

Reply:
I think the point of this test is to test the `(lsp_label + lsp_label_detail).startswith(lsp_filter_text)` logic so the filter text should be either like now or have extra characters at the end.

Reply:
Does that clarify or am I wrong here?

Reply:
actually will remove that test since it appears to be basically the same as the one above it, just less clear on what is expected.